src/base/bittorrent/session.cpp

```diff
@@ -5158,7 +5158,8 @@ void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
     const auto calcRate = [interval](const qint64 previous, const qint64 current) -> qint64
     {
         Q_ASSERT(current >= previous);
-        return ((current - previous) / interval);
```

Comment:
> Please explain the bug/why it doesn't work.

Apparently, some artifacts of floating-point division.
When numerator is 0 it can produces non-zero result with some denominators.

Reply:
>When numerator is 0 it can produces non-zero result with some denominators.

That doesn't make sense mathematically and therefore the code would look confusing (at least to me).

Can you try dumping the values? I would like to investigate myself but I don't know how to reproduce.

This could be related: https://stackoverflow.com/questions/38571860/can-division-by-non-zero-still-create-a-nan-infinity

Reply:
> Can you try dumping the values? I would like to investigate myself but I don't know how to reproduce.

OK, I'll do it as soon as I get to the machine where I can reproduce it.

Reply:
Wow, it turned out to be so successful that I can run tests with a lot of torrent even on a weak machine. So I did it. You were right. The problem is not in the numerator (it's just dependent), but in the denominator. If the timestamps turn out to be quite close, then we get a zero interval, which in turn leads to division by zero, which is acceptable with floating-point numbers.
Here is the modified code I used for testing:
```c++
{
    Q_ASSERT(current >= previous);
    const auto floatResult = (current - previous) / interval;
    const qint64 intResult = floatResult;
    if (intResult < 0)
    {
        qDebug() << "ALERT!!!";
        qDebug() << "previous =" << previous;
        qDebug() << "current =" << current;
        qDebug() << "interval =" << interval;
        qDebug() << "floatResult =" << floatResult;
        qDebug() << "intResult =" << intResult;
    }
    return intResult;
};
```
Here is one of the results:
>previous = 0
current = 0
interval = 0
floatResult = nan
intResult = -9223372036854775808

So I changed the fix as follows:
1. added a zero interval check, 
2. changed the resolution of the interval to microseconds (BTW, I no longer received a zero interval after that, but it's still better to have a zero check), 
3. stopped using floating-point calculations (IMO, they are not required here, since we still discard the fractional part in the end).